criu/sk-unix.c

```diff
@@ -1787,10 +1787,12 @@ static int open_unixsk_standalone(struct unix_sk_info *ui, int *new_fd)
 	 * into it in a special way.
 	 */
 	if (peer && (peer->flags & USK_GHOST_FDSTORE)) {
-		fle_peer = file_master(&peer->d);
-		if (fle_peer->stage < FLE_OPEN) {
+		/*
+		 * Here we return 1 which says "try again" to
+		 * the caller if peer is not yet bounded.
+		 */
+		if (peer->bound == 0)
```

Comment:
Probably it should be "if (peer_is_not_prepared(peer)) return 1;".

Reply:
Yes, technically it will be more correct. I've also tried to reproduce deadlock with STREAM unix socket, but the design on interaction with stream sockets implies that we open "connection" sockets *after* server socket. And this circumstance is very important and breaks possible deadlock reproduce. I even put a comment in zdtm test which says that to reproduce a deadlock we need to have a situation where peer socket is the *last* file descriptor of the holding process. I've also tried to "move" file descriptor using `dup2` but it doesn't help me to reproduce the problem.

Reply:
Fixed